### PR TITLE
OJ-39182 Mark custom field issue mismatches for redownload

### PR DESCRIPTION
**Pull request by ashleyredzko**
This uses the `mark_for_redownload` kwarg in `identify_custom_field_mismatches` to send mismatched custom field issues to Jellyfish to redownload them.

### CODE REVIEW

File: jf_agent/jf_jira/jira_download.py
Comment (lheureuxe13) [inline jf_agent/jf_jira/jira_download.py:126]: Nit: could you run `pdm run black jf_agent/jf_jira/jira_download.py` to reformat a few long lines of code in this change?

**Review by lheureuxe13 — COMMENTED** [review]
Same question as Jack on your other PR re: logging in jf_ingest, otherwise LGTM

File: jf_agent/jf_jira/jira_download.py
Comment (ashleyredzko) [inline jf_agent/jf_jira/jira_download.py:135]: TODO: If https://github.com/Jellyfish-AI/jf_ingest/pull/192 is pulled into this PR, this is redundant and can be removed.

File: jf_agent/jf_jira/jira_download.py
Comment (ashleyredzko) [inline jf_agent/jf_jira/jira_download.py:80]: This logging is either redundant or completely ported over to `jf_ingest`.

File: jf_agent/jf_jira/jira_download.py
Comment (ashleyredzko) [inline jf_agent/jf_jira/jira_download.py:170]: All of the above logging is now present in `jf_ingest` version `0.0.138`, and is redundant here.

**Review by lheureuxe13 — APPROVED** [review]
Might want to wait until the morning to merge so that we can monitor more closely, otherwise LGTM 🫡

**Comment by jruel4** [comment]
> Might want to wait until the morning to merge so that we can monitor more closely, otherwise LGTM 🫡

+1 otherwise LGTM

**Review by jruel4 — CHANGES_REQUESTED** [review]
Couple of things

File: pdm.lock
Comment (jruel4) [inline pdm.lock:52]: Q: Any way to do targeted update of only `jf_ingest`? Maybe I'm misremembering, but this feels like more changes than usual.

File: jf_agent/jf_jira/jira_download.py
Comment (jruel4) [inline jf_agent/jf_jira/jira_download.py:63]: B: This functionality needs to be retained. We only get the issue IDs to redownload at the beginning (whenever we initially generate `IngestionConfig`). Without this functionality, we won't pick up issues that are marked for redownload until the next agent run.

**Review by jruel4 — APPROVED** [review]
Excellent, thank you!

**Pull request merged**
